srml/support/src/dispatch.rs

```diff
@@ -67,38 +67,122 @@ pub trait Parameter: Codec + Clone + Eq {}
 #[cfg(not(feature = "std"))]
 impl<T> Parameter for T where T: Codec + Clone + Eq {}
 
-/// Declare a module struct and implement the dispatch logic.
+/// Declares a `Module` struct and a `Call` enum, which implements the dispatch logic.
 ///
-/// Usually used as follows:
+/// ## Declaration
 ///
+/// ```
+/// # #[macro_use]
+/// # extern crate srml_support;
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, Trait, ensure_signed};
+/// decl_module! {
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+/// 		fn my_function(origin, var: u64) -> Result {
```

Comment:
Would like to see a comment here about what a private function within `decl_module!` represents.